sdk/scheduler/src/main/java/com/mesosphere/sdk/metrics/Metrics.java

```diff
@@ -145,4 +151,53 @@ public static void record(Protos.TaskStatus taskStatus) {
         final String metricName = String.format("task_status.%s", taskStatus.getState().name().toLowerCase());
         metrics.counter(metricName).inc();
     }
+
+    public static void setPlanStatus(Optional<String> namespace, String planName, Status status) {
+        final String metricName = namespace.isPresent() ?
+                String.format("plan_status.%s.%s", namespace.get(), planName)
+                :
+                String.format("plan_status.%s", planName);
+
+        Map<String, Gauge> gauges = metrics.getGauges((name, metric) -> name.equals(metricName));
+        if (gauges.size() > 0) {
+            ((PlanGauge) gauges.get(metricName)).setStatus(status);
+        } else {
+            PlanGauge gauge = new PlanGauge().setStatus(status);
+            metrics.gauge(metricName, () -> gauge);
+        }
+    }
+
+    static class PlanGauge implements Gauge<Integer> {
+        private Status status;
+
+        public PlanGauge setStatus(Status status) {
+            this.status = status;
+            return this;
+        }
+
+        @Override
+        public Integer getValue() {
+            // We return a set of user facing statuses rather than the complete set of statuses available.
+            // This also insulates us against changes to the Status enum.
+            //
+            // The values step upwards when the plan is active.
+            switch (status) {
+                case ERROR:
+                    return -1;
+                case COMPLETE:
+                    return 0;
+                case WAITING:
+                case PENDING:
+                    return 1;
+                case IN_PROGRESS:
+                case PREPARED:
+                case STARTED:
+                case STARTING:
+                    return 2;
+                default:
+                    throw new IllegalStateException(
```

Comment:
Will this crash the application? Would we not want this included in the error class? Alternatively we can log it and add `-99` or something?

Reply:
Alternatively, we should have a test in place that *ENSURES* that we never publish code that could have an invalid state.